lib/ansible/modules/cloud/azure/azure_rm_managed_disk.py

```diff
@@ -353,6 +372,9 @@ def is_different(self, found_disk, new_disk):
         if new_disk.get('disk_size_gb'):
             if not found_disk['disk_size_gb'] == new_disk['disk_size_gb']:
                 resp = True
+        if new_disk.get('os_type'):
+            if not found_disk['os_type'] == new_disk['os_type']:
```

Comment:
It is not possible to remove the os_type based on my testing.  The enumeration has no "undefined" setting, so once it is set, it cannot be unset, only changed.  I'll look at changing this to a self.fail instead.

Reply:
The scenario you describe is intentional.  If the user does not specify the os_type, we do nothing to the field.  This is because os_type cannot be removed, only set.